src/euphorie/client/browser/session.py

```diff
@@ -329,8 +329,15 @@ def setupSession(self):
             # Touch means: the modification timestamp is set.
             # But we need to make sure the refreshed marker is up to date!
             survey_session.refresh_survey(survey)
-            return survey_session
+            new_session = survey_session
+        else:
+            new_session = self.rebuild_session(survey, profile)
 
+        new_session.update_measure_types(survey)
```

Comment:
I‌ looks a bit heavy to me to update all measure types for all risks every time one edits the Profile...
Would a upgrade step be better?
Not sure if this is worth the effort - it depends on how often the Profile edit view is used.

Reply:
An upgrade step is meant to run once. We need code that runs every time a tool is changed and republished (specifically when an `in_place_standard` measure is deleted.

> I‌ looks a bit heavy to me to update all measure types for all risks every time one edits the Profile...

Please note that `profile` here refers to the profile questions on a tool. They determine which optional modules are active in the respective session. The profile view checks whether the structure of the tool has changed, because this might mean that the structure of the session needs to be adjusted.

We're not updating all measure types. The query specifically filters for `plan_type == in_place_standard`, and the type is only changed if a measure has been deleted.

I think this is pretty lightweight compared to the [`treeChanges`](https://github.com/euphorie/Euphorie/blob/97b14f3c41f3b9dc6b89e2641647115fa7500b24/src/euphorie/client/update.py#L90) method that runs every time the profile view is called, and especially the [rebuilding of the session](https://github.com/euphorie/Euphorie/blob/97b14f3c41f3b9dc6b89e2641647115fa7500b24/src/euphorie/client/browser/session.py#L334) that happens when `treeChanges` detects a change.

Reply:
Thanks for the insight!